lib/tesla/multipart.ex

```diff
@@ -103,7 +103,7 @@ defmodule Tesla.Multipart do
         false -> headers
       end
 
-    data = File.stream!(path, [], 2048)
```

Comment:
This changes resolves the dialyzer errors:

```
lib/tesla/multipart.ex:94: Function add_file/2 has no local return
lib/tesla/multipart.ex:94: Function add_file/3 has no local return
lib/tesla/multipart.ex:106: The call 'Elixir.File':'stream!'(_path@1::binary() | maybe_improper_list(binary() | maybe_improper_list(any(),binary() | []) | char(),binary() | []),[],2048) breaks the contract ('Elixir.Path':t(),'line' | pos_integer(),[stream_mode()]) -> 'Elixir.File.Stream':t()
```

But as mentioned in #642, I'm not sure how to make this change in a backwards compatible way. See the soft deprecations section of https://github.com/elixir-lang/elixir/releases/tag/v1.16.0